metals/src/main/scala/scala/meta/metals/lsp/MetalsService.scala

```diff
@@ -0,0 +1,79 @@
+package scala.meta.metals.lsp
+
+import java.util
+import java.util.concurrent.CompletableFuture
+
+import scala.meta.internal.metals.DidFocusResult
+import scala.meta.internal.metals.WindowStateDidChangeParams
+import scala.meta.internal.metals.doctor.DoctorVisibilityDidChangeParams
+import scala.meta.internal.metals.findfiles.FindTextInDependencyJarsRequest
+import scala.meta.internal.tvp.MetalsTreeViewChildrenResult
+import scala.meta.internal.tvp.TreeViewChildrenParams
+import scala.meta.internal.tvp.TreeViewNodeCollapseDidChangeParams
+import scala.meta.internal.tvp.TreeViewNodeRevealResult
+import scala.meta.internal.tvp.TreeViewParentParams
+import scala.meta.internal.tvp.TreeViewParentResult
+import scala.meta.internal.tvp.TreeViewVisibilityDidChangeParams
+
+import org.eclipse.lsp4j._
+import org.eclipse.lsp4j.jsonrpc.services.JsonNotification
+import org.eclipse.lsp4j.jsonrpc.services.JsonRequest
+
+/**
+ * Interface which describes metals specific LSP requests and notifications which are
+ * implemented by Metals.
+ *
+ * Each method has a default implementation which throws UnsupportedOperationException. Throwing in this context is ok because:
+ * - the default implementation should never be called, because the method is always overridden in [[scala.meta.internal.metals.MetalsLanguageServer]]
```

Comment:
Out of curiosity, do these even need a default implementation then? Couldn't they just all be abstract?